servers/physics_3d/shape_3d_sw.cpp

```diff
@@ -1980,7 +1984,7 @@ void HeightMapShape3DSW::set_data(const Variant &p_data) {
 		PackedByteArray im_data = image->get_data();
 		heights_buffer.resize(image->get_width() * image->get_height());
 
-		float *w = heights_buffer.ptrw();
+		real_t *w = heights_buffer.ptrw();
```

Comment:
You should double check places where you changed `float *` pointers to `real_t *` pointers, as it's likely places, like here, where we then cast a buffer pointer and size/alignment matters.

See here, `real_t *w` won't be compatible with `float *rp`.

Reply:
The `ERR_FAIL_MSG` below is no longer valid since you may expect PackedFloat32Array.